pkg/client/imagestreams.go

```diff
@@ -108,3 +115,30 @@ func (c *imageStreams) UpdateStatus(stream *imageapi.ImageStream) (result *image
 	err = c.r.Put().Namespace(c.ns).Resource("imageStreams").Name(stream.Name).SubResource("status").Body(stream).Do().Into(result)
 	return
 }
+
+// Import makes a call to the server to retrieve information about the requested images or to perform an import. ImageStreamImport
+// will be returned if no actual import was requested (the to fields were not set), or an ImageStream if import was requested.
+func (c *imageStreams) Import(isi *imageapi.ImageStreamImport) (*imageapi.ImageStreamImport, error) {
+	result := &imageapi.ImageStreamImport{}
+	if err := c.r.Post().Namespace(c.ns).Resource("imageStreamImports").Body(isi).Do().Into(result); err != nil {
+		return nil, transformUnsupported(err)
```

Comment:
There are lots of contexts where you don't have the discovery doc.  We're not going to require clients to be initialized with discovery docs in all cases (the things that use clients may not necessarily want to do discovery).  That might be an additional gate, but a check of this form is intended.

And there's a reason the interface is `error`, not `*unversioned.Status` - there are lots of reasons client calls can fail beyond the server telling them.  This is just another type of error.